### DISCUSSION TITLE
Only input tensors may be passed as positional arguments. The following argument value should be passed as a keyword argument: 1000 (of type <class 'int'>)

### DISCUSSION THREAD

**Discussion by kdimou** [NVlabs/sionna#388] [General]
I have installed Sionna by following the instructions in the page 

https://nvlabs.github.io/sionna/installation.html

I tried the steps suggested therein and installed python, TensorFlow and Sionna.

Did the test 

python
```py
import sionna

print(sionna.__version__)
0.16.2
```

and the sionna version that appears is indeed 0.16.2

Tried to execute locally the "Hello World" by launching the jupyterLab via

```
python -m jupyterlab .
```

and the following message appears:

**`ValueError: Only input tensors may be passed as positional arguments. The following argument value should be passed as a keyword argument: 1000 (of type <class 'int'>)`**

It seems that there is an error in my environment and the tensorFlow package is not uploaded? I tried to execut the command locally at a terminal by using python (and python 3) and the same error appears.

```py
import sionna
>>> binary_source = sionna.utils.BinarySource()
>>> b = binary_source([batch_size, num_bits_per_symbol])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/konstantinosdimou/Simulators/SionnaNewMacbook/.venv/lib/python3.11/site-packages/keras/src/utils/traceback_utils.py", line 122, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/Users/konstantinosdimou/Simulators/SionnaNewMacbook/.venv/lib/python3.11/site-packages/keras/src/layers/layer.py", line 721, in __call__
    raise ValueError(
ValueError: Only input tensors may be passed as positional arguments. The following argument value should be passed as a keyword argument: 1000 (of type <class 'int'>)
```

It seems that either Tensorflow is not installed properly, or another package is missing, or the downloaded TensorFlow is not visible in my environment?

(Of course, I can run the examples and tutorials at Google Colab)

**Comment by merlinND** [comment]
Hello @kdimou,

My first guess would be about an incompatible TensorFlow version. Could you please check which version you have installed?
According to `requirements.txt`, the highest currently supported version is 2.15
https://github.com/NVlabs/sionna/blob/7eb04f43356333386b4361361188e3810620a0d8/requirements.txt#L1

**Reply by kdimou** [reply]
Hello Merlin,

Indeed, this solved the problem. The Tensorflow version installed was the version 2.16.1. I created a virtual environment, where I installed tensorflow 2.15. Within this virtual environment I launched jupyterlab and the code now can be executed.
Below are the commands executed.

```bash
pip show tensorflow # (for checking the tensorflow version)
python3 -m venv tf215_env # (definition of virtual environment)
source tf215_env/bin/activate # (activation of virtual environment)
pip install tensorflow==2.15 # (installation of the Tensorflow version 2.15 within the virtual environment)
pip install jupyter notebook ipykernel # (installed jupyter notebook in the virtual environment)
jupyter notebook # (jupyter notebook launching)
```
